src/library_wasm_worker.js

```diff
@@ -145,6 +145,10 @@ addToLibrary({
   _emscripten_create_wasm_worker__postset: `
 if (ENVIRONMENT_IS_WASM_WORKER) {
   _wasmWorkers[0] = this;
+#if AUDIO_WORKLET
+  // Audio Worklets do not have postMessage()ing capabilities.
+  if (!ENVIRONMENT_IS_AUDIO_WORKLET)
```

Comment:
(Audio Worklets do have postMessage()ing capabilities, but those capabilities occur via a message port, and not via a global scope object, so the comment is not quite accurate)

However, here the statement `_wasmWorkers[0] = this;` in an Audio Worklet is unnecessary. Technically the statement would have to be `_wasmWorkers[0] = globalThis['messagePort'];`, but that statement needs to be located elsewhere, so no need to worry about that here.

So maybe this would be cleanest to read

```js
  _emscripten_create_wasm_worker__postset: `
if (ENVIRONMENT_IS_WASM_WORKER
#if AUDIO_WORKLET
  && !ENVIRONMENT_IS_AUDIO_WORKLET
#endif
  ) {
  _wasmWorkers[0] = this;
  addEventListener("message", _wasmWorkerAppendToQueue);
}
```
This way both of these statements that don't relate to audio worklets will be skipped.

Reply:
Just pushed a commit with this suggestion + more accurate comment. Thanks for taking a careful look at this!